legion/jupyterlab-plugin/src/commands/cloud.ts

```diff
@@ -0,0 +1,307 @@
+/**
+ *   Copyright 2019 EPAM Systems
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+import { showErrorMessage, showDialog, Dialog } from '@jupyterlab/apputils';
+import { ReadonlyJSONObject } from '@phosphor/coreutils';
+
+import { CommandIDs, IAddCommandsOptions } from './base';
+import * as dialogs from '../components/dialogs';
+import * as models from '../models/cloud';
+import * as cloudDialogs from '../components/dialogs/cloud';
+
+export interface ICloudScaleParameters {
+
+}
+
+/**
+ * Add the commands for the legion extension.
+ */
+export function addCommands(options: IAddCommandsOptions) {
+    let { commands } = options.app;
+
+    commands.addCommand(CommandIDs.newCloudTraining, {
+        label: 'Train model in a cloud',
+        caption: 'Create new cloud training',
+        execute: args => {
+
+            let values = (args as unknown) as cloudDialogs.ICreateNewTrainingDialogValues;
+
+            cloudDialogs.showCreateNewTrainingDialog(options.apiState.cloud.vcss, values)
+                .then(({ value, button }) => {
+                    if (button.accept) {
```

Comment:
Can you refactor this file? For example, extract some code to functions.
I find it very difficult to read.